components/scripts/gradle/03-validate-local-build-caching-different-locations.sh

```diff
@@ -158,8 +170,33 @@ execute_second_build() {
 }
 
 fetch_build_cache_metrics() {
-  read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  if [ "$build_scan_publishing_mode" == "on" ]; then
+    read_build_scan_metadata
+    fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  else
+    find_build_scan_dumps
+    read_build_scan_dumps
+  fi
+}
+
+find_build_scan_dumps() {
+  local first_build_dir="${EXP_DIR}/first-build_${project_name}"
+  first_build_scan_dump="$(find "$first_build_dir" -maxdepth 1 -regex '^.*build-scan.*\.scan$')"
```

Comment:
I'd like some feedback on this approach. I couldn't really find a better way.

Reply:
+1: I think this approach is fine.

The other way we could do this is with `ls -l | grep PATTERN`, but grep behaves differently on different flavors of Unix (for example, the version of grep MacOS ships with is less capable than the one that ships with most distributions of Linux).  In contrast, the `find` command's behavior is more consistent across platforms.

-1: I do suggest adding `-type f` before `-regex` to the find command though. That will limit find to only looking at files and not directories as well.  

It may also be possible to write a more specific regex based on the pattern `build-scan-%s-%s-%d-%s.scan` Something like `^.*build-scan-.*-.*-.*-.*\.scan`, which will further reduce the chances that we pick up a random file in the user's repo.